control-plane/csi-driver/src/bin/controller/pvwatcher.rs

```diff
@@ -101,8 +103,39 @@ impl PvGarbageCollector {
     /// 2. to tackle k8s bug where volumes are leaked when PV deletion is attempted before
     /// PVC deletion.
     async fn delete_orphan_volumes(&self) {
-        let max_entries = 200;
+        let volumes = self.collect_volume_ids().await;
+        if volumes.is_empty() {
+            return;
+        }
+        let Some(pvcs) = self.collect_pvc_ids().await else {
+            return;
+        };
+
+        let mut gc_uids = Vec::with_capacity(volumes.len());
+        for volume_uid in volumes {
+            if self.is_vol_orphan(volume_uid, &pvcs).await {
+                gc_uids.push(volume_uid);
+            }
+        }
+        tokio::time::sleep(std::time::Duration::from_secs(2)).await;
```

Comment:
Do we need to sleep here?

Reply:
It's just to avoid interfering with the regular CSI deletion, by giving it some time.

Reply:
Should we wait a little longer? Maybe 15 or 20 seconds?